126 restyle

CollinsSpencer opened the pull request:
This PR is massive.

There's still work that needs to be done, but it requires LOTS of testing, so I wanted to get what I have on here.

Still needs to be done:
* Restyle public queue
* Restyle private queue
* Add content to the volunteer page
* Add content to the management page
* Add content to the footer

I also archived each of the index.php files to a neighboring index_old.php file. These will need to be removed before this PR is merged, but adding the contents to the template file was much easier with the reference. 

Again, this will require LOTS of testing.

Testing:
- [x] Volunteer (still needs content)
  - [x] Title, breadcrumbs
  - [x] Contents
- [x] Admin/Management (still needs content)
  - [x] Title, breadcrumbs
  - [x] Contents
  - [x] Correct access/privilege handling
- [x] Navigation
- [x] Home Page
  - [x] Title, breadcrumbs
  - [x] Contents
- [x] Questionnaire
  - [x] Title, breadcrumbs
  - [x] Contents
- [x] Signup
  - [x] Title, breadcrumbs
  - [x] Contents
- [x] Login
  - [x] Title, breadcrumbs
  - [x] Contents
- [x] Profile
  - [x] Title, breadcrumbs
  - [x] Contents
  - [x] Correct access/privilege handling
- [x] Manage Users
  - [x] Title, breadcrumbs
  - [x] Contents
  - [x] Correct access/privilege handling
- [x] Print Documents
  - [x] Title, breadcrumbs
  - [x] Contents
  - [x] Correct access/privilege handling
- [x] 401/unauthorized
  - [x] Title, breadcrumbs
  - [x] Contents
- [x] 404
  - [x] Title, breadcrumbs
  - [x] Contents
- [x] 500
  - [x] Title, breadcrumbs
  - [x] Contents
- [x] Public Queue
  - [x] Title, breadcrumbs
  - [x] Contents
- [x] Private Queue
  - [x] Title, breadcrumbs
  - [x] Contents
  - [x]  Correct access/privilege handling

MatthewMeacham commented:
You. are. a. god.

MatthewMeacham commented:
This is absolutely incredible.

MatthewMeacham commented:
One thing I did notice is that the logout option in the navigation menu--when you hover over it, instead of being a pointer, it was a text cursor which is sort of odd.

MatthewMeacham commented:
Another small bug, hitting save when editing personal information on the /profile page--the changes are persisted database-wise, but not shown to the user unless they refresh (or hit edit->cancel). I think this is a wrong id somewhere in the JS.

MatthewMeacham commented:
The /management page should probably have some authorization/permission checks around it. Like if the person doesn't have the use_admin_tools permission, then it should send them to the unauthorized page

MatthewMeacham commented on index.template.php:
```diff
@@ -0,0 +1,118 @@
+<?php
```
What's with this file? Is this what you were basing the rest of the pages off of?

MatthewMeacham commented on management/documents/index.php:
```diff
@@ -1,38 +1,140 @@
+					<form class="wdn-inner-wrapper wdn-inner-padding-no-top wdn-set-grid">
+						<div class="bp768-wdn-col-one-half">
+							<label for="dateInput">Date</label>
+							<input type="text" id="dateInput"/>
```
Thanks for switching this over to a nice datapicker thing

MatthewMeacham commented on management/users/index.php:
```diff
@@ -1,85 +1,167 @@
-	<script src="/dist/management/users/users.js"></script>
+	<?php wdnInclude("/wdn/templates_4.1/includes/body_scripts.html"); ?>
+	<?php require_once "$root/server/global_includes.php"; ?>
+	<script src="/management/users/users.js"></script>
```
We're missing `/dist/...` here

MatthewMeacham commented on php.fixed.dwt.php:
```diff
@@ -0,0 +1,115 @@
+<?php
```
Or was this your template?

MatthewMeacham commented on profile/index.php:
```diff
@@ -1,85 +1,187 @@
-				<input type="text" id="emailInput" style="display:none;" disabled />
-			</div>
+							<div>
+								<label for="firstName" id="firstNameLabel">First Name:</label>
```
Should this label now be for `firstNameText`?

MatthewMeacham commented on server/callbacks.php:
```diff
@@ -26,7 +26,7 @@ function getLoginClass(){
 	require_once "$root/server/Login.class.php";
 
-	return new Login($DB_CONN, 'VITA', 'VITA', '/index.php', '/register/index.php', 'noreply@vita.unl.edu', 'vita@cse.unl.edu');
+	return new Login($DB_CONN, 'VITA', 'VITA', '/', '/register/index.php', 'noreply@vita.unl.edu', 'vita@cse.unl.edu');
```
thank you for doing this, I've meant to do this for a while.

MatthewMeacham commented on management/users/index.php:
```diff
@@ -1,85 +1,167 @@
+<!-- <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" /> -->
+<!-- <link rel="stylesheet" href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap-glyphicons.css" /> -->
+<!-- <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/css/bootstrap-select.min.css" /> -->
+<link rel="stylesheet" href="/management/users/users.css" />
```
missing the `/dist/...` here as well

MatthewMeacham commented on questionnaire/index.php:
```diff
@@ -1,202 +1,322 @@
+<title>Questionnaire | VITA Lincoln | University of Nebraska&ndash;Lincoln</title>
+<!-- TemplateEndEditable -->
+<!-- TemplateBeginEditable name="head" -->
+<link rel="stylesheet" href="/dist/assets/css/bootstrap.btn-group.min.css">
```
Should we have the `/dist` here or nah?

CollinsSpencer replied:
I don't think it hurts.

CollinsSpencer commented:
I downloaded ```php.fixed.dwt.php``` straight from the [UNL WDN site](https://wdn.unl.edu/starting-site) in ```v 4.1 Templates```. I made some changes to it that were needed for every page and saved it as ```index.template.php```.

Both will be deleted eventually. ```php.fixed.dwt.php``` can be used for reference, like the formatting of the <title> tag. ```index.template.php``` can be used for making any new pages. That's why I haven't deleted them yet.

CollinsSpencer commented:
Also, I think we might want some additional validation around the phone number in the profile page. I can currently set the phone number to be "a" if want. It might work to just use something like this jQuery plugin. [http://digitalbush.com/projects/masked-input-plugin/](http://digitalbush.com/projects/masked-input-plugin/)

Thoughts?

MatthewMeacham commented:
@CollinsSpencer I have no problems adding validation around the profile page phone number. I suppose more than likely, volunteers will have US phone numbers. But something to consider is that on the appt signup, we also have a required phone number field, but international students probably don't have US phone numbers, so validating against a US phone number wouldn't work. 

I suppose volunteers will probably have US phone numbers, but what if they don't? What then?

CollinsSpencer commented:
@MatthewMeacham 
These are the files that need to be deleted:
- ```/components/nav.php```
- ```/error_pages/404_old.php```
- ```/error_pages/500_old.php```
- ```/login/index_old.php```
- ```/management/documents/index_old.php```
- ```/management/users/index_old.php```
- ```/profile/index_old.php```
- ```/questionnaire/index_old.php```
- ```/queue/index_old.php```
- ```/queue/private_old.php```
- ```/queue/public_old.php```
- ```/register/index_old.php```
- ```/signup/index_old.php```
- ```/unauthorized/index_old.php```
- ```/index_old.php```
- ```/index.template.php```
- ```/php.fixed.dwt.php```

Should I go through and delete all of them right now? I think we should probably leave ```/index.template.php``` and delete the rest.

MatthewMeacham commented:
@CollinsSpencer Feel free to delete those files. They're in source control if we need them again

MatthewMeacham commented on profile/profile.js:
```diff
@@ -1,395 +1,419 @@
+	function appendSignedUpShift(siteTitle, dateString, startTimeString, endTimeString, userShiftId, siteId, roleName) {
+		let shiftRow = $('<div></div>');
+		let shiftInformation = $('<span></span>').html(`${siteTitle}: ${dateString} ${startTimeString} - ${endTimeString} (${roleName})`);
+		let removeButton = $('<i></i>').addClass('fa fa-trash-o icon clickable').click(function() {
```
I did notice that there appears to be no trash can icon in the HTML, probably because you removed font awesome. What are our choices for getting this remove button back?

CollinsSpencer replied:
It doesn't appear that they have a trash can icon. [https://wdn.unl.edu/documentation/icons](https://wdn.unl.edu/documentation/icons)

I'll look into this further.

MatthewMeacham commented on queue/index.php:
```diff
@@ -1,12 +1,131 @@
+			<div id="maincontent" class="wdn-main">
+				<div id="pagetitle">
+					<!-- TemplateBeginEditable name="pagetitle" -->
+					<h1>Queue</h1>
```
Having it say "Queue" here looks kinda odd... should we remove this? Although it would ruin the consistency

CollinsSpencer replied:
I think it would be fine to remove just the line containing ```<h1>Queue</h1>```.

CollinsSpencer replied:
Done.

MatthewMeacham commented on queue/queuePrivateController.js:
```diff
@@ -0,0 +1,62 @@
+			$scope.client.ended = true;
+			let urlSafeExplanation = fixedEncodeURIComponent(explanation);
+			QueueDataService.incompleteAppointment(urlSafeExplanation, $scope.client.appointmentId);
+			$('textarea').val('');
```
This isn't working on my machine

CollinsSpencer replied:
Done.

MatthewMeacham commented on queue/private.php:
```diff
@@ -1,141 +1,164 @@
+				<button class="wdn-button wdn-button-brand" 
+					ng-show="!client.checkedIn" 
+					ng-click="cancelledAppointment()">
+					Submit Cancelled Appointment
```
I just realized that one can hit this button a bunch of times and it will create a new entry in ServicedAppointment each time someone hits this button... I suppose we could add a line into the PHP that ensures there isn't already an entry? But perhaps we should (perhaps in addition) add in something on the front-end that prevents someone from hitting this button if the appointment has ended? Same thing with the incompleteAppointment button?

CollinsSpencer replied:
Definitely. I'll look into it.

CollinsSpencer replied:
Done.

JakeKo commented:
A few comment:
* Break the line between permissions in Admin/User Management
* When I navigate to Admin/Print Documents, the date field is not initially occupied
* When I download either of the Excel spreadsheets from Admin/Print Documents, the date field is reset to blank
* Logout doesn't work (the issue is complex enough I'd prefer to talk about it in person)

Beyond these __minor__ problems, the styling looks phenomenal. Mad props Scollins.
